### PR TITLE
rcal-700 Update exposure pipeline to correct cal_step and suffixes

### LINKED ISSUE #962: Update elpp for tweakreg returning an empty modelcontainer
_Issue [RCAL-700](https://jira.stsci.edu/browse/RCAL-700) was created on JIRA by [David Davis](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ddavis):_

When tweakreg is set to skip in the pipeline it returns an empty ModelContainer. When the model is "popped" from the container the operation fails. We need to update the logic to test and account for that in the exposure pipeline.

### PR DESCRIPTION
**Pull request by ddavis-stsci**
Resolves [RCAL-700](https://jira.stsci.edu/browse/rcal-700)

Closes #962

This PR 
For SDF the first stage of the pipeline will run and produce
```
run_elpp.csh test_asn.json
...
2023-11-03 10:02:28,955 - stpipe.ExposurePipeline.tweakreg - INFO - Step skipped.
2023-11-03 10:02:28,956 - stpipe.ExposurePipeline.tweakreg - INFO - Step tweakreg done
2023-11-03 10:02:28,956 - stpipe.ExposurePipeline - INFO - Roman exposure calibration pipeline ending...
2023-11-03 10:02:32,525 - stpipe.ExposurePipeline - INFO - Saved model in r0000101001001001001_01101_0001_WFI01_sourcedetection.asdf
2023-11-03 10:02:35,976 - stpipe.ExposurePipeline - INFO - Saved model in r0000101001001001001_01101_0001_WFI02_sourcedetection.asdf
2023-11-03 10:02:38,760 - stpipe.ExposurePipeline - INFO - Saved model in r0000101001001001001_01101_0001_WFI03_sourcedetection.asdf
2023-11-03 10:02:38,760 - stpipe.ExposurePipeline - INFO - Step ExposurePipeline done
```

with the cal_step for tweakreg set to
`            tweakreg	SKIPPED`
The second step will run tweakreg on the individual files produced and yield _cal files
```
run_tweakreg.csh test_tweakreg_asn.json
...
2023-11-03 10:21:09,236 - stpipe.TweakRegStep - INFO - Saved model in r0000101001001001001_01101_0001_WFI01_cal.asdf
2023-11-03 10:21:13,649 - stpipe.TweakRegStep - INFO - Saved model in r0000101001001001001_01101_0001_WFI02_cal.asdf
2023-11-03 10:21:18,120 - stpipe.TweakRegStep - INFO - Saved model in r0000101001001001001_01101_0001_WFI03_cal.asdf
2023-11-03 10:21:18,120 - stpipe.TweakRegStep - INFO - Step TweakRegStep done
```

with 
with the cal_step for tweakreg set to
`            tweakreg	COMPLETE`

**Checklist**
- [x] added entry in `CHANGES.rst` under the corresponding subsection
- [x] updated relevant tests
- [N/A] updated relevant documentation
- [x] updated relevant milestone(s)
- [x] added relevant label(s)
- [N/A] ran regression tests, post a link to the Jenkins job below. [How to run regression tests on a PR](https://github.com/spacetelescope/romancal/wiki/Running-Regression-Tests-Against-PR-Branches)
Regression testing is currently broken, local tests pass

### CODE REVIEW

File: romancal/pipeline/exposure_pipeline.py [hunk 1]
```diff
@@ -171,8 +174,12 @@ def process(self, input):
                 result.meta.cal_step.photom = "SKIPPED"
                 result.meta.cal_step.source_detection = "SKIPPED"
                 result.meta.cal_step.tweakreg = "SKIPPED"
+                self.suffix = "COMPLETE"
```
Comment (schlafly) [inline romancal/pipeline/exposure_pipeline.py:177]: This looks weird?  Do we really want to set the file name suffix to COMPLETE?

File: romancal/pipeline/exposure_pipeline.py [hunk 2]
```diff
@@ -183,24 +190,46 @@ def process(self, input):
...
-                result = mc_result._models.pop()
+                if hasattr(ModelContainer(result), "_models") and mc_result._models:
+                    result = mc_result._models.pop()
+                else:
+                    result.meta.cal_step.tweakreg == "SKIPPED"
+
             if file_type == "asn":
                 result = self.tweakreg(tweakreg_input)
+                for model in result._models:
+                    model.meta.cal_step.tweakreg = "SKIPPED"
```
Comment (schlafly) [inline romancal/pipeline/exposure_pipeline.py:204]: I think I expected COMPLETE here?

**Review by schlafly — APPROVED** [review]
Looks good to me.  There were a couple of lines that surprised me that I noted inline.

File: romancal/pipeline/exposure_pipeline.py (hunk 2 above)
Comment (ddavis-stsci) [reply romancal/pipeline/exposure_pipeline.py:204]: When tweakreg is run with the option "--steps.tweakreg.skip=True" it is not honoring that and setting the step to skipped. So for the test RCAL-665 to pass this needs to be manually set to "SKIPPED".

This should not remain in place. My plan is to write a function that will check the skip flag that is passed to the step and set it appropriately and use this for all the pipeline steps. I think we should have a uniform method. I checked JWST and they seem to handle this flag in a non-uniform manner and I don't want to reproduce that. I'd like to fix this after B12.

Comment (schlafly) [reply romancal/pipeline/exposure_pipeline.py:204]: Sounds good, thanks.

**Pull request merged**